plugin/evm/tx_test.go

```diff
@@ -180,3 +185,145 @@ func executeTxTest(t *testing.T, test atomicTxTest) {
 		test.checkState(t, vm)
 	}
 }
+
+func TestEVMOutputLess(t *testing.T) {
+	type test struct {
+		name     string
+		a, b     EVMOutput
+		expected bool
+	}
+
+	tests := []test{
+		{
+			name: "first address less",
+			a: EVMOutput{
+				Address: common.BytesToAddress([]byte{0x01}),
+				AssetID: ids.ID{1},
+			},
+			b: EVMOutput{
+				Address: common.BytesToAddress([]byte{0x02}),
+				AssetID: ids.ID{0},
+			},
+			expected: true,
+		},
+		{
+			name: "first address greater",
+			a: EVMOutput{
+				Address: common.BytesToAddress([]byte{0x02}),
+				AssetID: ids.ID{0},
+			},
+			b: EVMOutput{
+				Address: common.BytesToAddress([]byte{0x01}),
+				AssetID: ids.ID{1},
+			},
+			expected: false,
+		},
+		{
+			name: "addresses equal; first assetID less",
+			a: EVMOutput{
+				Address: common.BytesToAddress([]byte{0x01}),
+				AssetID: ids.ID{0},
+			},
+			b: EVMOutput{
+				Address: common.BytesToAddress([]byte{0x01}),
+				AssetID: ids.ID{1},
+			},
+			expected: true,
+		},
+		{
+			name: "addresses equal; first assetID greater",
+			a: EVMOutput{
+				Address: common.BytesToAddress([]byte{0x01}),
+				AssetID: ids.ID{1},
+			},
+			b: EVMOutput{
+				Address: common.BytesToAddress([]byte{0x01}),
+				AssetID: ids.ID{0},
+			},
+			expected: false,
+		},
+		{
+			name:     "equal",
+			a:        EVMOutput{},
+			b:        EVMOutput{},
+			expected: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require.Equal(t, tt.expected, tt.a.Less(tt.b))
+		})
+	}
+}
+
+func TestEVMInputLess(t *testing.T) {
+	type test struct {
+		name     string
+		a, b     EVMInput
+		expected bool
+	}
+
+	tests := []test{
+		{
+			name: "first address less",
+			a: EVMInput{
+				Address: common.BytesToAddress([]byte{0x01}),
+				AssetID: ids.ID{1},
+			},
+			b: EVMInput{
+				Address: common.BytesToAddress([]byte{0x02}),
+				AssetID: ids.ID{0},
+			},
+			expected: true,
+		},
+		{
+			name: "first address greater",
+			a: EVMInput{
+				Address: common.BytesToAddress([]byte{0x02}),
+				AssetID: ids.ID{0},
+			},
+			b: EVMInput{
+				Address: common.BytesToAddress([]byte{0x01}),
+				AssetID: ids.ID{1},
+			},
+			expected: false,
+		},
```

Comment:
```suggestion
		{
			name: "first address less",
			a: EVMInput{
				Address: common.BytesToAddress([]byte{0x01}),
				AssetID: ids.ID{},
			},
			b: EVMInput{
				Address: common.BytesToAddress([]byte{0x02}),
				AssetID: ids.ID{},
			},
			expected: true,
		},
		{
			name: "first address greater",
			a: EVMInput{
				Address: common.BytesToAddress([]byte{0x02}),
				AssetID: ids.ID{},
			},
			b: EVMInput{
				Address: common.BytesToAddress([]byte{0x01}),
				AssetID: ids.ID{},
			},
			expected: false,
		},
```

Could we make the assetID's equal in these test cases so that it's clear the only difference is the address?

Reply:
I think it makes sense to keep it as is to show that the address is the first thing checked for when comparing inputs/outputs. If you want I could add another case where the assetIDs are equal.

Reply:
Sure let's add one more case for this

Reply:
added